src/clients/Client.cc

```diff
@@ -455,33 +455,34 @@ sameUrlHosts(const char *url1, const char *url2)
 static void
 purgeEntriesByHeader(HttpRequest *req, const char *reqUrl, Http::Message *rep, Http::HdrType hdr)
 {
-    const char *hdrUrl, *absUrl;
-
-    absUrl = NULL;
-    hdrUrl = rep->header.getStr(hdr);
-    if (hdrUrl == NULL) {
+    const char *hdrUrl;
+    if (!(hdrUrl = rep->header.getStr(hdr)))
```

Comment:
Please simplify, improve const-correctness, and reduce code duplication:

```suggestion
    const auto hdrUrl = rep->header.getStr(hdr);
    if (!hdrUrl)
```

The `const` part would require small changes in the code below, but I think all these changes combined actually make that (rather dangerous) code clearer and safer.